tutorials/ab3TracingCode.md

```diff
@@ -88,15 +89,45 @@ Bingo\! `MainWindow#executeCommand()` seems to be exactly what we’re looking f
 Now let’s set the breakpoint. First, double-click the item to reach the corresponding code. Once there, click on the left gutter to set a breakpoint, as shown below.
  ![LeftGutter](images/tracing/LeftGutter.png)
 
+
 ## Tracing the execution path
 
-Recall from the User Guide that the `edit` command has the format: `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [t/TAG]…` For this tutorial we will be issuing the command `edit 1 n/Alice Yeoh`.
+At this point, you should have appreciated the general sequence diagram [**shown above**](#before-we-start)
+
+
+For this code tracing,
+we will explore the `EditCommand` which is a feature that allows users to edit exising person fields in AB3.
+
+<box type="tip" seamless>
+    Recall from the <a href="https://se-education.org/addressbook-level3/UserGuide.html#editing-a-person--edit">User Guide</a> that the <code>edit</code> command has the format: <code>edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [t/TAG]…</code>.
+</box>
+
+For this tutorial, we will be issuing the command <code>edit 1 n/Alice Yeoh</code>
+
+Specifically,
+we trace the following sequence diagram
+which explores the `Logic` component in more detail which will be paramount to understanding other commands as well!
+The diagrams will be reproduced with labels in their sections to highlight the pass of control between classes.
+
+<puml src="images/tracing/LogicSequenceDiagramWithMainWindow.puml" alt="Tracing an `edit` command with specific arguments"></puml>
+
 
 <box type="tip" seamless>
 
 **Tip:** Over the course of the debugging session, you will encounter every major component in the application. Try to keep track of what happens inside the component and where the execution transfers to another component.
 </box>
 
+
+
+
+
+### 1. MainWindow -> LogicManager
+
+<annotate src="images/tracing/LogicSequenceDiagramWithMainWindow.png" width="900" alt="Sample Image">
+  <!-- Set Legend to both -->
+  <a-point x="24%" y="16%"  label="1"/>
```

Comment:
Should we use a different label e.g., `t1` to avoid confusing with step numbers in the explanations?

Reply:
Sure will call it T1